### PR TITLE
Fix the EM importance sampling

### PR DESCRIPTION
**Pull request by ProfFan**
This PR should fix the EM importance sampling.

### CODE REVIEW

File: Sources/BeeTracking/TrackingLikelihoodModel.swift [hunk 1]
```diff
@@ -56,6 +56,19 @@ public struct TrackingLikelihoodModel<Encoder: AppearanceModelEncoder, FG:Genera
               foregroundModel : FG(from: fgFeatures, given:p?.foregroundModel),
               backgroundModel : BG(from: bgFeatures, given:p?.backgroundModel))
   }
+  
+  /// Calculate the negative log likelihood of the likelihood model of a patch
+  @differentiable
+  public func negativeLogLikelihood(of patch: Tensor<Double>) -> Double {
+    let encoded = encoder.encode(patch.expandingShape(at: 0)).squeezingShape(at: 0)
```
Comment (dellaert) [inline Sources/BeeTracking/TrackingLikelihoodModel.swift:63]: Works in batch?

File: Sources/BeeTracking/TrackingLikelihoodModel.swift [hunk 2]
```diff
@@ -101,7 +114,16 @@ extension TrackingLikelihoodModel : McEmModel {
       // sample from noise model on manual pose
       var proposal = datum.obb.center
       proposal.perturbWith(stddev: Vector3(0.3, 8, 4.6))
-      return (1.0, proposal)
+      return (
+        // The weight of the proposal, which equals to the unnormalized probability of the likelihood:
+        self.unnormalizedProbability(
```
Comment (dellaert) [inline Sources/BeeTracking/TrackingLikelihoodModel.swift:119]: I’d prefer to use a variable for the patch

**Review by dellaert — APPROVED** [review]

File: Sources/BeeTracking/TrackingLikelihoodModel.swift (hunk 2 above)
Comment (ProfFan) [reply Sources/BeeTracking/TrackingLikelihoodModel.swift:119]: Sure, I'll do that!

File: Sources/BeeTracking/TrackingLikelihoodModel.swift (hunk 1 above)
Comment (ProfFan) [reply Sources/BeeTracking/TrackingLikelihoodModel.swift:63]: @dellaert I am thinking about changing this interface from `encode()` to `encode(batch: )` and `encode(sample: )`, how is that?

Comment (ProfFan) [reply Sources/BeeTracking/TrackingLikelihoodModel.swift]: done

**Pull request merged**